### PR TITLE
refactor(drive): remove duplicated withdrawal amount validation

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

- Withdrawal amount is validated two times.
- Missing test coverage for withdrawal ST validation

## What was done?

- Remove addition amount validation in the validate state function

## How Has This Been Tested?

Add test for balance and structure validation

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced a validation mechanism for identity credit withdrawal transitions, ensuring compliance with specified limits and criteria.
	- Added comprehensive unit tests for the new validation logic.

- **Bug Fixes**
	- Enhanced error handling for insufficient balance during identity credit withdrawals.

- **Refactor**
	- Streamlined balance validation logic by removing unnecessary checks and optimizing calculations.
	- Updated validation logic for output scripts and pooling methods.

- **Tests**
	- Added tests for various scenarios in the identity credit withdrawal process to ensure robustness.

### CODE REVIEW

File: packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ impl InvalidIdentityCreditWithdrawalTransitionOutputScriptError {
     }
 
     pub fn output_script(&self) -> CoreScript {
+        // TODO: We shouldn't clone in getter 🤦
         self.output_script.clone()
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs:33]: yeah, but I don't want to change it in this PR

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/structure/v1/mod.rs [hunk 2]
```diff
@@ -78,3 +78,230 @@ impl IdentityCreditWithdrawalStateTransitionStructureValidationV1
...
+            let platform_version = &PLATFORM_V1;
+
+            let result = transition
+                .validate_basic_structure_v1(platform_version)
+                .expect("failed to validate basic structure");
+
+            assert!(result.is_valid());
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/structure/v1/mod.rs]: Good point!

File: packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs [hunk 3]
```diff
@@ -29,6 +29,7 @@ impl InvalidIdentityCreditWithdrawalTransitionOutputScriptError {
     }
 
     pub fn output_script(&self) -> CoreScript {
+        // TODO: We shouldn't clone in getter 🤦
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs:32]: It's fine for an error.

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/state/v0/mod.rs [hunk 4]
```diff
@@ -91,7 +65,7 @@ impl IdentityCreditWithdrawalStateTransitionStateValidationV0
     ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error> {
         let consensus_validation_result =
             IdentityCreditWithdrawalTransitionAction::try_from_identity_credit_withdrawal(
-                &platform.drive,
+                platform.drive,
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/state/v0/mod.rs:68]: are you sure we can remove the &?

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs (hunk 3 above)
Comment (shumkov) [reply packages/rs-dpp/src/errors/consensus/basic/identity/invalid_identity_credit_withdrawal_transition_output_script_error.rs:32]: It doesn't matter what it is. Hidden clone is a bad practice.

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/state/v0/mod.rs (hunk 4 above)
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_credit_withdrawal/state/v0/mod.rs:68]: Yes, clippy warning

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**